kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -326,6 +326,11 @@ protected TransactionImpl buildTransactionInternal(
         throw new UnsupportedOperationException(
             "REPLACE TABLE is not yet supported with column mapping");
       }
+      if (newProtocol.orElse(baseProtocol).supportsFeature(TableFeatures.ROW_TRACKING_W_FEATURE)) {
```

Comment:
I see 5 usages of `newProtocol.orElse(baseProtocol)` and 10 usages of `newMetadata.orElse(baseMetadata)`

In another PR, could you please create `txnProtocol` and `txnMetadata` private helper methods?

Reply:
Hmm both `newProtocol` and `baseProtocol` are local variables I'm not sure the best way to refactor this now (without majorly refactoring how we build the transaction)?